invokeai/app/invocations/controlnet_image_processors.py

```diff
@@ -476,37 +479,86 @@ def run_processor(self, image: Image.Image) -> Image.Image:
+        if ksize % 2 == 0:
+            ksize += 1  # ksize must be odd
+        blurred_image = cv2.GaussianBlur(image_np, (ksize, ksize), sigmaX=sigma_x)
+        return blurred_image
+
+    # referenced from
+    # https://huggingface.co/TTPlanet/TTPLanet_SDXL_Controlnet_Tile_Realistic/blob/37f1c4575b543fb2036e39f5763d082fdd135318/TTP_tile_preprocessor_v5.py
+    def _apply_guided_filter(self, image_np: np.ndarray[Any, Any], radius: int, eps: float, scale: int):
+        filter = FastGuidedFilter(image_np, radius, eps, scale)
+        return filter.filter(image_np)
+
+    def _regular_resample(self, np_img: np.ndarray[Any, Any]):
+        height, width, _ = np_img.shape
         np_img = HWC3(np_img)
-        if down_sampling_rate < 1.1:
+        if self.down_sampling_rate < 1.1:
             return np_img
-        H, W, C = np_img.shape
-        H = int(float(H) / float(down_sampling_rate))
-        W = int(float(W) / float(down_sampling_rate))
-        np_img = cv2.resize(np_img, (W, H), interpolation=cv2.INTER_AREA)
+
+        new_height = int(float(height) / float(self.down_sampling_rate))
+        new_width = int(float(width) / float(self.down_sampling_rate))
+        np_img = cv2.resize(np_img, (new_width, new_height), interpolation=cv2.INTER_AREA)
+        return np_img
+
+    # referenced from
+    # https://huggingface.co/TTPlanet/TTPLanet_SDXL_Controlnet_Tile_Realistic/blob/37f1c4575b543fb2036e39f5763d082fdd135318/TTP_tile_preprocessor_v5.py
+    def _blur_resample(self, np_img: np.ndarray[Any, Any]):
+        height, width, _ = np_img.shape
+        ratio = np.sqrt(1024.0 * 1024.0 / (width * height))
+        resize_w, resize_h = int(width * ratio), int(height * ratio)
+        np_img = cv2.resize(np_img, (resize_w, resize_h))
+
+        blur_strength = random.sample([i / 10.0 for i in range(10, 201, 2)], k=1)[0]
+        radius = random.sample([i for i in range(1, 40, 2)], k=1)[0]  # noqa: C416
+        eps = random.sample([i / 1000.0 for i in range(1, 101, 2)], k=1)[0]
+        scale_factor = random.sample([i / 10.0 for i in range(10, 181, 5)], k=1)[0]
+
+        if random.random() > 0.5:
+            np_img = self._apply_gaussian_blur(np_img, ksize=int(blur_strength), sigma_x=blur_strength / 2)
+
+        if random.random() > 0.5:
+            np_img = self._apply_guided_filter(np_img, radius, eps, int(scale_factor))
+
+        np_img = cv2.resize(
+            np_img, (int(resize_w / scale_factor), int(resize_h / scale_factor)), interpolation=cv2.INTER_AREA
+        )
+        np_img = cv2.resize(np_img, (resize_w, resize_h), interpolation=cv2.INTER_CUBIC)
+        return np_img
+
+    def _super_resample(self, np_img: np.ndarray[Any, Any]):
+        height, width, _ = np_img.shape
+        ratio = np.sqrt(1024.0 * 1024.0 / (width * height))
+        resize_w, resize_h = int(width * ratio) // 48 * 48, int(height * ratio) // 48 * 48
+        np_img = cv2.resize(np_img, (resize_w, resize_h))
```

Comment:
Same question as above re: why 1024x1024?

And why force it to be a multiple of 48?